Source/WTF/wtf/text/StringImpl.h

```diff
@@ -409,27 +409,30 @@ class StringImpl : private StringImplShape {
 
     // FIXME: Do these functions really belong in StringImpl?
     template<typename CharacterType>
-    ALWAYS_INLINE static void copyCharacters(CharacterType* destination, std::span<const CharacterType> source)
+    ALWAYS_INLINE static void copyCharacters(std::span<CharacterType> destination, std::span<const CharacterType> source)
     {
-        return copyElements(destination, source.data(), source.size());
+        RELEASE_ASSERT(destination.size() >= source.size());
```

Comment:
Seems like this can go inside `copyElements` instead if we change it so it gets a destination size argument either separately or as part of moving it to use `span` arguments.

Reply:
I agree, eventually. The thing is that this is a larger change (since there are more callers of `copyElements()`) and it is even more likely to negatively impact perf.